machine-learning/02-openpbta_consensus_clustering.Rmd

```diff
@@ -0,0 +1,326 @@
+
+```{r subtype_df}
+mb_molecular_subtype_df <- histologies_df %>%
+  filter(short_histology == "Medulloblastoma") %>%
+  select(Kids_First_Biospecimen_ID, molecular_subtype)
+```
+
+Add the consensus clustering labels.
+
+```{r subtype_cc, live = TRUE}
+# Create a data frame that contains the consensus cluster results and join
+# it with the data frame of molecular subtype labels
+cc_df <- data.frame(cc_cluster_labels) %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  inner_join(mb_molecular_subtype_df)
+```
+
+Do the consensus clustering results agree with the molecular subtype labels?
+
+```{r cc_subtype_table}
+table(cc_df$cc_cluster_labels, cc_df$molecular_subtype)
+```
+
+Hm... there's some agreement with the subtype labels but it's not perfect. 
+Why might that be?
+Let's start by looking at the overview figure for the medulloblastoma classifier.
+
+![](https://github.com/d3b-center/medullo-classifier-package/raw/aba25b6acc0f8dab11f46db6955a38ffac3d0140/images/workflow.png)
+
+We used a different measure (VST values vs. FPKM) and used all features. 
+
+## Low-dimensional representation
+
+Let's see if low-dimensional representations agree with our other unsupervised results.
+First, let's find the high variance genes; we'll perform dimension reduction on those genes only.
+
+```{r gene_variance}
+# Calculate variance
+gene_variance <- matrixStats::rowVars(rnaseq_mat)
+# Find the value that we'll use as a threshold to filter the top 5%
+variance_threshold <- quantile(gene_variance, 0.95)
+# Row indices of high variance genes
+high_variance_index <- which(gene_variance > variance_threshold)
+```
+
+### UMAP
+
+First, let's use UMAP.
+
+```{r run_umap}
+# Set seed for reproducible UMAP results
+set.seed(2020)
+# umap() expects features (genes) to be columns, so we have to use t()
+umap_results <- umap::umap(t(rnaseq_mat[high_variance_index, ]))
+```
+
+The UMAP coordinates are in the `layout` element of the list returned by `umap::umap()`.
+
+```{r make_umap_df, live = TRUE}
+# Make a data frame of the layout results and join with molecular subtype 
+# data frame
+umap_plot_df <- data.frame(umap_results$layout) %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  inner_join(mb_molecular_subtype_df)
+```
+
+Let's make a scatter plot and color our samples by the subtype labels.
+
+```{r umap_scatter}
+umap_plot_df %>%
+  ggplot(aes(x = X1, 
+             y = X2,
+             color = molecular_subtype)) +
+  geom_point() +
+  colorblindr::scale_color_OkabeIto() +
+  theme_bw() +
+  xlab("UMAP1") +
+  ylab("UMAP2")
+```
+
+### PCA 
+
+Now, we'll briefly show you how to use built-in functions for PCA on any matrix that's not in a specialized object for some kind of genomic data.
+
+```{r run_pca}
+# Like umap(), prcomp() expects the features to be columns
+pca_results <- prcomp(t(rnaseq_mat[high_variance_index, ]), 
+                      scale = TRUE)
+# The low-dimensional representation is returned in x
```

Comment:
I was looking at this in the rendered notebook, so while this is not my full review, I noticed this comment mostly because the `head()` still prints a whole lot. Technically, PCA has the same dimensionality, we just throw some away after calculating them. `calculatePCA()` does this by default, so that may be what they expect, but you might want to mention that `prcomp()` does not (unless you set `tol`), so the output matrix has the same dimensions as the input. 

At least that is my understanding!

Reply:
In any case, you might want to use `pca_results[1:6, 1:6]` or something similar to reduce the printout.

Reply:
Sounds good 👍  `calculatePCA()` returns the first 50 PCs based on my read.